service/de_encoding/secretKeyCreator.go

```diff
@@ -2,42 +2,12 @@ package de_encoding
 
 import (
 	"crypto/sha256"
-	"fmt"
-	"io/ioutil"
-	"math/rand"
 )
 
-func GenerateRandomBytes(n int) ([]byte, error) {
-	b := make([]byte, n)
-	_, err := rand.Read(b)
-	if err != nil {
-		return nil, err
-	}
-	return b, nil
-}
-
-func shaEncrypt(fileNameFunc string) []byte {
+func PassToSecretKey(pass string) []byte {
 	h := sha256.New()
-	h.Write([]byte(fileNameFunc))
-	encryptedKeyWord := h.Sum(nil)
-	return encryptedKeyWord
-}
-
-func CheckFile(name string) []byte {
-	randomKey, _ := GenerateRandomBytes(32)
-	fileName := shaEncrypt(name)
-	data, err := ioutil.ReadFile(string(fileName))
-	if err != nil {
-		fmt.Println(err)
-	}
-	if data != nil {
-		return data
-	} else {
-		err := ioutil.WriteFile(string(fileName), randomKey, 0777)
+	h.Write([]byte(pass))
+	encryptedKeyWord := (h.Sum(nil))
+	return encryptedKeyWord[:32]
```

Comment:
Why 32? What is that number? Be more intentional and descriptive, especially for constants. 

And what will happen if `encryptedKeyWord` has less length than 32 bytes length? How to handle it properly? Please, rewrite.

Reply:
> Why 32? What is that number? Be more intentional and descriptive, especially for constants.
> 
> And what will happen if `encryptedKeyWord` has less length than 32 bytes length? How to handle it properly? Please, rewrite.

changed it in the [last commit](https://github.com/retttddd/golangProjectv3/pull/1/commits/71bf236e00c46fafeb2c823e500a06b7c0eb6862)

Reply:
in the last commit reduced the whole algorithm and added panic